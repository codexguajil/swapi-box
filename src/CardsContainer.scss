#cards-container {
    background: hsl(0, 0%, 100%);
    color: hsl(0, 0%, 0%);
    display: flex;
    flex-wrap: wrap;
    font-size: 40px;
    margin: auto;
    overflow: scroll;
    position: absolute;
    right: 50px;
    top: 250px;
    height: 570px;
    width: 69%;
}