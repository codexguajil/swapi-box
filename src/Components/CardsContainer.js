import React from 'react';
import Card from './Card';
import propTypes from 'prop-types';

export const CardsContainer = ({data}) => {
    return (
            <div id="cards-container">
            {data.length > 0 &&
                data.map(data => {
                    return <Card data={data}/>
                })
            }
            </div>
    )
}

CardsContainer.propTypes = {
    data: propTypes.array.isRequired
}