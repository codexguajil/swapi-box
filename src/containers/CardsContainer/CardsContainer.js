import React, { Component } from 'react';
import Card from '../../Components/Card';
import { connect } from 'react-redux';

export class CardsContainer extends Component {
  
  render() {
    const { people, clickedButton } = this.props
    return (
            <div id="cards-container">
              {
                clickedButton === 'people' &&
                 people.map((person,i) => {
                  return <Card data={person} key={i} />
                })
              }
            </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  people: state.people,
  clickedButton: state.clickedButton,
})

export default connect(mapStateToProps)(CardsContainer);