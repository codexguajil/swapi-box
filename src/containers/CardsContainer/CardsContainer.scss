#cards-container {
    background: lightblue;
    color: hsl(0, 0%, 0%);
    display: flex;
    flex-wrap: wrap;
    font-size: 40px;
    margin: auto;
    overflow: scroll;
    position: absolute;
    right: 50px;
    top: 250px;
    height: 570px;
    width: 69%;
}