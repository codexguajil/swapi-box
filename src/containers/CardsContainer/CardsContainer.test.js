import React from 'react';
import {shallow} from 'enzyme';
import {CardsContainer} from './CardsContainer'; 

describe('CardsContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <CardsContainer />
    )
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})