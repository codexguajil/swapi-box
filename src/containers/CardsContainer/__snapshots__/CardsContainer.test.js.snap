// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsContainer should match snapshot 1`] = `
<div
  id="cards-container"
/>
`;
